ribbon/src/main/java/feign/ribbon/LoadBalancingTarget.java

```diff
@@ -45,26 +45,38 @@
 
   private final String name;
   private final String scheme;
+  private final String path;
   private final Class<T> type;
   private final AbstractLoadBalancer lb;
+  
+  @Deprecated
   protected LoadBalancingTarget(Class<T> type, String scheme, String name) {
     this.type = checkNotNull(type, "type");
     this.scheme = checkNotNull(scheme, "scheme");
     this.name = checkNotNull(name, "name");
+    this.path = "";
+    this.lb = AbstractLoadBalancer.class.cast(getNamedLoadBalancer(name()));
+  }
+  
+  protected LoadBalancingTarget(Class<T> type, String scheme, String name, String path) {
+    this.type = checkNotNull(type, "type");
+    this.scheme = checkNotNull(scheme, "scheme");
+    this.name = checkNotNull(name, "name");
```

Comment:
I didn't want to change existing implementation, but it seems like `url()` should actually return scheme + path, not name (according to `Target.java` docs)